core/src/test/java/kafka/log/remote/RemoteLogOffsetReaderTest.java

```diff
@@ -111,8 +111,11 @@ public void testTaskQueueFullAndCancelTask() throws Exception {
         holderList.get(2).jobFuture().cancel(false);
 
         rlm.resume();
-        AsyncOffsetReadFutureHolder<Either<Exception, Option<TimestampAndOffset>>> last = holderList.get(holderList.size() - 1);
-        last.taskFuture().get(100, TimeUnit.MILLISECONDS);
+        for (AsyncOffsetReadFutureHolder<Either<Exception, Option<TimestampAndOffset>>> holder : holderList) {
+            if (!holder.jobFuture().isCancelled()) {
+                holder.taskFuture().get(1, TimeUnit.SECONDS);
+            }
+        }
```

Comment:
OK, we now want to explicitly wait for all tasks completed because on CI, the machine is quite slow. Besides, I think we can also reduce the queue size to a smaller number, ex: 3, to use less resource to make the test more reliable.

Reply:
Reduce the queue size to 3.